src/jupyterlabcontroller/storage/k8s.py

```diff
@@ -674,3 +682,430 @@ def _standard_metadata(self, name: str) -> V1ObjectMeta:
                 "argocd.argoproj.io/sync-options": "Prune=false",
             },
         )
+
+    """It's really tempting to create generic methods for object
+    creation and deletion, that do some fancy exception handling to do
+    retries and ignore error-cases-that-are-normal-operation,
+    especially since Kubernetes methods are regular in both names and
+    argument type and ordering, and we want to do very similar things
+    for many objects.
+
+    How hard could it be, you think, to use getattr() to pluck the
+    right method, and then reuse the guts of creation/retry code?  And
+    then you realize that you could do this to await the creation of
+    an arbitrary Kubernetes object also.
+
+    I am here to warn you, dear reader, that this turns out to be a
+    terrible idea, because it turns errors you could have caught
+    quickly and easily with the type system into runtime errors deep
+    in the FastAPI event loop, with no obvious way to connect the
+    stack trace back to the place you forgot an "await" or misspelled
+    a method name."""
+
+    # Methods for fileserver
+
+    async def create_fileserver(
+        self,
+        username: str,
+        namespace: str,
+        job: V1Job,
+        service: V1Service,
+        gf_ingress: dict[str, Any],
+    ) -> bool:
+        with FILESERVER_LOCK:
+            self._logger.info(f"Creating new fileserver for {username}...")
+            self._logger.debug(f"...creating new job for {username}")
+            await self._create_fileserver_job(username, namespace, job)
+            self._logger.debug(f"...creating new service for {username}")
+            await self._create_fileserver_service(
+                username, namespace, spec=service
+            )
+            self._logger.debug(f"...creating new gfingress for {username}")
+            await self._create_fileserver_gafaelfawringress(
+                username, namespace, spec=gf_ingress
+            )
+            self._logger.debug(
+                f"...polling until objects appear for {username}"
+            )
+            timeout = 60.0
+            interval = 3.9
+            try:
+                async with asyncio.timeout(timeout):
+                    while True:
+                        good = await self.check_fileserver_present(
+                            username, namespace
+                        )
+                        if good:
+                            self._logger.info(
+                                f"Fileserver created for {username}"
+                            )
+                            return True
+                        await asyncio.sleep(interval)
```

Comment:
Polling is messy and will make startup unnecessarily slow. I think we should always use event watches instead of polling, since Kubernetes has very nice event watch support.

Reply:
That sounds like it's gonna be painful but, OK.

Reply:
I've done this for the Pod.  Haven't yet extended watches to the non-Pod objects.  It wasn't that painful.

Reply:
Going to resolve this.  More watch work needed, but that will happen on DM-38972